src/runtime_src/xdp/profile/plugin/aie_debug/aie_debug_plugin.cpp

```diff
@@ -0,0 +1,397 @@
+          << "Status 0x" << std::hex << stuckCoreStatus << std::dec
+          << " : " << getCoreStatusString(stuckCoreStatus);
+
+          xrt_core::message::send(severity_level::warning, "XRT", warningMessage.str());
+          foundStuckCores = false;
+        }
+
+        // Print status for debug
+        if (xrt_core::config::get_verbosity() >= static_cast<unsigned int>(severity_level::debug)) {
+          std::stringstream msg;
+          for (const auto& tile : graphTilesVec) {
+            if (coreStuckCountMap[tile]) {
+              msg
+                << "T(" << tile.col <<"," << tile.row + 1 << "):" << "<" << coreStuckCountMap[tile]
+                << ":0x" << std::hex << coreStatusMap[tile] << std::dec << "> ";
+            }
+          }
+          if (!msg.str().empty()) {
+            msg << std::endl << "Graph " << graphName << " #Cur : " << graphStallCounter << " #Thr : " << graphTilesVec.size();
+            xrt_core::message::send(severity_level::debug, "XRT", msg.str());
+          }
+        }
+      } // For graphs
+
+      // Always write out latest debug/status file
+      for (auto w : writers) {
+        w->write(false);
+      }
+
+      std::this_thread::sleep_for(std::chrono::microseconds(mPollingInterval));     
+    }
+  }
+
+  void AIEDebugPlugin::updateAIEDevice(void* handle)
+  {
+    // Don't update if no debug/status is requested
+    if (!xrt_core::config::get_aie_status())
+      return;
+
+    const unsigned int PATH_LENGTH = 512;
+    char pathBuf[PATH_LENGTH];
+    memset(pathBuf, 0, PATH_LENGTH);
+    xclGetDebugIPlayoutPath(handle, pathBuf, PATH_LENGTH);
+
+    std::string sysfspath(pathBuf);
+    uint64_t deviceId = db->addDevice(sysfspath); // Get the unique device Id
+
+    if (!(db->getStaticInfo()).isDeviceReady(deviceId)) {
+      // Update the static database with information from xclbin
+      (db->getStaticInfo()).updateDevice(deviceId, handle);
+      {
+        struct xclDeviceInfo2 info;
+        if(xclGetDeviceInfo2(handle, &info) == 0) {
+          (db->getStaticInfo()).setDeviceName(deviceId, std::string(info.mName));
+        }
+      }
+    }
+
+    // Update list of tiles to debug
+    getTilesForDebug(handle);
+
+    // Open the writer for this device
+    struct xclDeviceInfo2 info;
+    xclGetDeviceInfo2(handle, &info);
+    std::string deviceName = std::string(info.mName);
+    // Create and register writer and file
+    std::string outputFile = "aie_status_" + deviceName + ".json";
+
+    VPWriter* writer = new AIEDebugWriter(outputFile.c_str(),
+                                          deviceName.c_str(), mIndex);
+    writers.push_back(writer);
+    db->getStaticInfo().addOpenedFile(writer->getcurrentFileName(), "AIE_RUNTIME_STATUS");
+
+    // Start the AIE debug thread
+    mThreadCtrlMap[handle] = true;
+    auto device_thread = std::thread(&AIEDebugPlugin::pollAIERegisters, this, mIndex, handle);
+    mThreadMap[handle] = std::move(device_thread);
```

Comment:
Do not overcomplicate with `std::move` and do not use the cryptic `std::thread` API with pointers and args now we have beautiful lambdas.
```suggestion
    mThreadMap[handle] = std::thread { [=] { AIEDebugPlugin::pollAIERegisters(this, mIndex, handle); } } ;
```

Reply:
this doesn't compile. Also std::thread can't be assigned so it has to be moved.

Reply:
But the code could be tweaked to compile with a lambda, just that I have not tried to compile and fix the syntax, as I do not have the big picture of the code.
Perhaps just
```c++
mThreadMap[handle] = std::thread { [=] { pollAIERegisters(mIndex, handle); } } ;
```
Otherwise your sentence *Also std::thread can't be assigned so it has to be moved.* is not correct.
My point is: why going through a useless variable `device_thread` that requires the obscure `std::move` instead of just using the thread directly?
To keep the same code:
```c++
mThreadMap[handle] = std::thread(&AIEDebugPlugin::pollAIERegisters, this, mIndex, handle);
```
is not a copy assignment because the `std::thread` constructor returns a pure R-value (prvalue https://en.cppreference.com/w/cpp/language/value_category)
So it is a move assignment operation happening here.

Reply:
changed it. The lambda format does indeed look much better.